Mathlib/RepresentationTheory/GroupCohomology/LowDegree.lean

```diff
@@ -332,19 +347,285 @@ theorem oneCoboundaries_eq_bot_of_isTrivial (A : Rep k G) [A.IsTrivial] :
   simp_rw [oneCoboundaries, dZero_eq_zero]
   exact LinearMap.range_eq_bot.2 rfl
 
-theorem mem_twoCoboundaries_of_dOne_apply (x : G → A) :
-    ⟨dOne A x, LinearMap.ext_iff.1 (dTwo_comp_dOne A) x⟩ ∈ twoCoboundaries A :=
-  LinearMap.mem_range_self _ _
+/-- Makes a 2-coboundary out of `f ∈ Im(d¹)`. -/
+def twoCoboundariesOfMemRange (f : G × G → A) (h : f ∈ LinearMap.range (dOne A)) :
+    twoCoboundaries A :=
+  ⟨⟨f, LinearMap.range_le_ker_iff.2 (dTwo_comp_dOne A) h⟩,
+    by rcases h with ⟨x, rfl⟩; exact ⟨x, rfl⟩⟩
 
-theorem mem_twoCoboundaries_of_mem_range (f : G × G → A) (h : f ∈ LinearMap.range (dOne A)) :
-    ⟨f, LinearMap.range_le_ker_iff.2 (dTwo_comp_dOne A) h⟩ ∈ twoCoboundaries A := by
-  rcases h with ⟨x, rfl⟩; exact ⟨x, rfl⟩
+theorem twoCoboundariesOfMemRange_apply (f : G × G → A) {h : f ∈ LinearMap.range (dOne A)} :
+    (twoCoboundariesOfMemRange f h).1.1 = f := rfl
+
+/-- Makes a 2-coboundary out of `f : G × G → A` and `x : G → A` such that
+`ρ(g)(x(h)) - x(gh) + x(g) = f(g, h)` for all `g, h : G`. -/
+def twoCoboundariesOfEq (f : G × G → A) (x : G → A)
+    (hf : ∀ g h, A.ρ g (x h) - x (g * h) + x g = f (g, h)) :
+    twoCoboundaries A :=
+  twoCoboundariesOfMemRange f ⟨x, by ext g; exact hf g.1 g.2⟩
+
+theorem twoCoboundariesOfEq_apply (f : G × G  → A) (x : G → A)
+    {hf : ∀ g h, A.ρ g (x h) - x (g * h) + x g = f (g, h)} :
+    (twoCoboundariesOfEq f x hf).1.1 = f := rfl
 
 theorem mem_range_of_mem_twoCoboundaries (f : twoCocycles A) (h : f ∈ twoCoboundaries A) :
     (twoCocycles A).subtype f ∈ LinearMap.range (dOne A) := by
   rcases h with ⟨x, rfl⟩; exact ⟨x, rfl⟩
 
 end Coboundaries
+section IsCocycle
+section
+variable {G A : Type*} [Mul G] [AddCommGroup A] [SMul G A]
+
+/-- A function `f : G → A` satisfies the 1-cocycle condition if
+`f(gh) = g • f(h) + f(g)` for all `g, h : G`. -/
+def IsOneCocycle (f : G → A) : Prop := ∀ g h : G, f (g * h) = g • f h + f g
+
+theorem isOneCocycle_iff (f : G → A) :
+    IsOneCocycle f ↔ ∀ g h : G, f (g * h) = g • f h + f g := Iff.rfl
+
+/-- A function `f : G × G → A` satisfies the 2-cocycle condition if
+`f(gh, j) + f(g, h) = g • f(h, j) + f(g, hj)` for all `g, h : G`. -/
+def IsTwoCocycle (f : G × G → A) : Prop :=
+  ∀ g h j : G, f (g * h, j) + f (g, h) = g • (f (h, j)) + f (g, h * j)
+
+theorem isTwoCocycle_iff (f : G × G → A) :
+    IsTwoCocycle f ↔ ∀ g h j : G, f (g * h, j) + f (g, h) = g • (f (h, j)) + f (g, h * j) :=
+  Iff.rfl
+
+end
+section
+variable {G A : Type*} [Monoid G] [AddCommGroup A] [MulAction G A]
+
+theorem map_one_of_isOneCocycle (f : G → A) (hf : IsOneCocycle f) :
+    f 1 = 0 := by
+  have := (isOneCocycle_iff f).1 hf 1 1
+  simpa only [mul_one, one_smul, self_eq_add_right] using this
```

Comment:
```suggestion
  simpa only [mul_one, one_smul, self_eq_add_right] using (isOneCocycle_iff f).1 hf 1 1
```
untested. If it works you can also remove the `by`. Same below.

Reply:
This works, thank you. Confused about the `by` comment, though

Reply:
Does `simpa only [mul_one, one_smul, self_eq_add_right] using hf 1 1` work?